In [58]:
import pandas as pd
import plotly.express as px
pd.set_option('display.max_columns', None)


In [59]:
df= pd.read_csv('/content/base_dados_projeto_lol.csv')
df.head()

/tmp/ipykernel_5694/1783834369.py:1: DtypeWarning:

Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.



,gameid,datacompleteness,url,league,year,split,playoffs,date,game,patch,participantid,side,position,playername,playerid,teamname,teamid,firstPick,champion,ban1,ban2,ban3,ban4,ban5,pick1,pick2,pick3,pick4,pick5,gamelength,result,kills,deaths,assists,teamkills,teamdeaths,doublekills,triplekills,quadrakills,pentakills,firstblood,firstbloodkill,firstbloodassist,firstbloodvictim,team kpm,ckpm,firstdragon,dragons,opp_dragons,elementaldrakes,opp_elementaldrakes,infernals,mountains,clouds,oceans,chemtechs,hextechs,dragons (type unknown),elders,opp_elders,firstherald,heralds,opp_heralds,void_grubs,opp_void_grubs,firstbaron,barons,opp_barons,atakhans,opp_atakhans,firsttower,towers,opp_towers,firstmidtower,firsttothreetowers,turretplates,opp_turretplates,inhibitors,opp_inhibitors,damagetochampions,dpm,damageshare,damagetakenperminute,damagemitigatedperminute,damagetotowers,wardsplaced,wpm,wardskilled,wcpm,controlwardsbought,visionscore,vspm,totalgold,earnedgold,earned gpm,earnedgoldshare,goldspent,gspd,gpr,total cs,minionkills,monsterkills,monsterkillsownjungle,monsterkillsenemyjungle,cspm,goldat10,xpat10,csat10,opp_goldat10,opp_xpat10,opp_csat10,golddiffat10,xpdiffat10,csdiffat10,killsat10,assistsat10,deathsat10,opp_killsat10,opp_assistsat10,opp_deathsat10,goldat15,xpat15,csat15,opp_goldat15,opp_xpat15,opp_csat15,golddiffat15,xpdiffat15,csdiffat15,killsat15,assistsat15,deathsat15,opp_killsat15,opp_assistsat15,opp_deathsat15,goldat20,xpat20,csat20,opp_goldat20,opp_xpat20,opp_csat20,golddiffat20,xpdiffat20,csdiffat20,killsat20,assistsat20,deathsat20,opp_killsat20,opp_assistsat20,opp_deathsat20,goldat25,xpat25,csat25,opp_goldat25,opp_xpat25,opp_csat25,golddiffat25,xpdiffat25,csdiffat25,killsat25,assistsat25,deathsat25,opp_killsat25,opp_assistsat25,opp_deathsat25
0,ESPORTSTMNT01_2690210,complete,NaN,LCKC,2022,Spring,0,2022-01-10 07:44:08,1,12.01,1,Blue,top,Soboro,oe:player:38e0af7278d6769d0c81d7c4b47ac1e,HANJIN BRION Challengers,oe:team:fa3d687a87bcae80362f784a7da571d,1.0,Renekton,Karma,Caitlyn,Syndra,Thresh,Lulu,NaN,NaN,NaN,NaN,NaN,1713,0,2,3,2,9,19,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.3152,0.9807,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,0.0,15768.0,552.2942,0.278784,1072.3993,777.7933,1528.0,8.0,0.2802,6.0,0.2102,5.0,26.0,0.9107,10934,7164.0,250.9282,0.253859,10275.0,NaN,NaN,231.0,220.0,11.0,NaN,NaN,8.0911,3228.0,4909.0,89.0,3176.0,4953.0,81.0,52.0,-44.0,8.0,0.0,0.0,0.0,0.0,0.0,0.0,5025.0,7560.0,135.0,4634.0,7215.0,121.0,391.0,345.0,14.0,0.0,1.0,0.0,0.0,1.0,0.0,6506.0,9853.0,172.0,6338.0,10200.0,171.0,168.0,-347.0,1.0,0.0,1.0,1.0,0.0,2.0,0.0,8462.0,11754.0,212.0,7857.0,12279.0,203.0,605.0,-525.0,9.0,0.0,1.0,1.0,0.0,2.0,0.0
1,ESPORTSTMNT01_2690210,complete,NaN,LCKC,2022,Spring,0,2022-01-10 07:44:08,1,12.01,2,Blue,jng,Raptor,oe:player:637ed20b1e41be1c51bd1a4cb211357,HANJIN BRION Challengers,oe:team:fa3d687a87bcae80362f784a7da571d,1.0,Xin Zhao,Karma,Caitlyn,Syndra,Thresh,Lulu,NaN,NaN,NaN,NaN,NaN,1713,0,2,5,6,9,19,0.0,0.0,0.0,0.0,1.0,0.0,1.0,0.0,0.3152,0.9807,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,1.0,11765.0,412.0841,0.208009,944.2732,650.1576,2511.0,6.0,0.2102,18.0,0.6305,6.0,48.0,1.6813,9138,5368.0,188.0210,0.190220,8750.0,NaN,NaN,148.0,33.0,115.0,NaN,NaN,5.1839,3429.0,3484.0,58.0,2944.0,3052.0,63.0,485.0,432.0,-5.0,1.0,2.0,0.0,0.0,0.0,1.0,5366.0,5320.0,89.0,4825.0,5595.0,100.0,541.0,-275.0,-11.0,2.0,3.0,2.0,0.0,5.0,1.0,6854.0,7193.0,116.0,6708.0,8275.0,142.0,146.0,-1082.0,-26.0,2.0,3.0,2.0,1.0,5.0,1.0,8254.0,8958.0,135.0,7833.0,9861.0,163.0,421.0,-903.0,-28.0,2.0,4.0,2.0,1.0,5.0,1.0
2,ESPORTSTMNT01_2690210,complete,NaN,LCKC,2022,Spring,0,2022-01-10 07:44:08,1,12.01,3,Blue,mid,Feisty,oe:player:d1ae0e2f9f3ac1e0e0cdcb86504ca77,HANJIN BRION Challengers,oe:team:fa3d687a87bcae80362f784a7da571d,1.0,LeBlanc,Karma,Caitlyn,Syndra,Thresh,Lulu,NaN,NaN,NaN,NaN,NaN,1713,0,2,2,3,9,19

##  Definição do Escopo e Recorte da Análise

Para que a nossa análise exploratória seja estatisticamente válida e taticamente comparável, precisamos isolar o ruído da base de dados original. O ecossistema competitivo global de League of Legends é composto por dezenas de ligas e diferentes divisões de base (Academy/Challengers).

Neste bloco, definimos os nossos alvos de estudo:
1. **Ligas Alvo:** `CBLOL` (Brasil), `LCK` (Coreia do Sul) e `LEC` (Europa), representando três culturas de jogo completamente distintas.
2. **Splits Alvo:** Selecionamos exclusivamente o `Split 2` e o `Summer Split` da temporada de 2022. Este recorte é crucial porque a segunda metade do ano reflete o auge tático das equipas, onde os elencos estão consolidados após a janela de transferências do meio do ano e o foco total é a qualificação para o *World Championship* (Mundial).

In [60]:
ligas_alvo = ['CBLOL' , 'LCK' , 'LEC']
splits_alvo = ['Split 2' , 'Summer']

###  Filtragem por Sumário de Equipas (`position == 'team'`)

Uma particularidade do formato de dados fornecido pelo *Oracle's Elixir* é que cada mapa gera 12 linhas: 10 para o desempenho individual de cada jogador e 2 linhas que consolidam as estatísticas globais das equipas (Lado Azul e Lado Vermelho).

Para as nossas métricas de macrofutebol (como tempo total, abates totais e placar de visão da equipa), precisamos de usar apenas as linhas consolidadas de equipa. Caso contrário, se usássemos as linhas dos jogadores, estaríamos a inflacionar e a quintuplicar os dados de tempo e abates do mapa. Filtramos isso definindo `position == 'team'` e aplicamos o método `.copy()` para evitar avisos de *SettingWithCopyWarning* no Pandas durante as transformações futuras.

In [61]:
df_filtrado = df.query("league in @ligas_alvo and split in @splits_alvo and position == 'team'").copy()

### Conversão de Tempo



In [62]:
df_filtrado['gamelength'] = df['gamelength'] / 60

In [63]:
df_filtrado.head()

,gameid,datacompleteness,url,league,year,split,playoffs,date,game,patch,participantid,side,position,playername,playerid,teamname,teamid,firstPick,champion,ban1,ban2,ban3,ban4,ban5,pick1,pick2,pick3,pick4,pick5,gamelength,result,kills,deaths,assists,teamkills,teamdeaths,doublekills,triplekills,quadrakills,pentakills,firstblood,firstbloodkill,firstbloodassist,firstbloodvictim,team kpm,ckpm,firstdragon,dragons,opp_dragons,elementaldrakes,opp_elementaldrakes,infernals,mountains,clouds,oceans,chemtechs,hextechs,dragons (type unknown),elders,opp_elders,firstherald,heralds,opp_heralds,void_grubs,opp_void_grubs,firstbaron,barons,opp_barons,atakhans,opp_atakhans,firsttower,towers,opp_towers,firstmidtower,firsttothreetowers,turretplates,opp_turretplates,inhibitors,opp_inhibitors,damagetochampions,dpm,damageshare,damagetakenperminute,damagemitigatedperminute,damagetotowers,wardsplaced,wpm,wardskilled,wcpm,controlwardsbought,visionscore,vspm,totalgold,earnedgold,earned gpm,earnedgoldshare,goldspent,gspd,gpr,total cs,minionkills,monsterkills,monsterkillsownjungle,monsterkillsenemyjungle,cspm,goldat10,xpat10,csat10,opp_goldat10,opp_xpat10,opp_csat10,golddiffat10,xpdiffat10,csdiffat10,killsat10,assistsat10,deathsat10,opp_killsat10,opp_assistsat10,opp_deathsat10,goldat15,xpat15,csat15,opp_goldat15,opp_xpat15,opp_csat15,golddiffat15,xpdiffat15,csdiffat15,killsat15,assistsat15,deathsat15,opp_killsat15,opp_assistsat15,opp_deathsat15,goldat20,xpat20,csat20,opp_goldat20,opp_xpat20,opp_csat20,golddiffat20,xpdiffat20,csdiffat20,killsat20,assistsat20,deathsat20,opp_killsat20,opp_assistsat20,opp_deathsat20,goldat25,xpat25,csat25,opp_goldat25,opp_xpat25,opp_csat25,golddiffat25,xpdiffat25,csdiffat25,killsat25,assistsat25,deathsat25,opp_killsat25,opp_assistsat25,opp_deathsat25
76030,ESPORTSTMNT02_2778045,complete,NaN,CBLOL,2022,Split 2,0,2022-06-11 15:46:13,1,12.1,100,Blue,team,NaN,NaN,paiN Gaming,oe:team:61450737f7b683db4da7b3e0bf0dced,1.0,NaN,Renata Glasc,Senna,Lucian,Swain,Ahri,Kalista,Gangplank,Nautilus,Nocturne,Galio,35.883333,0,17,23,49,17,23,2.0,0.0,0.0,0.0,1.0,NaN,NaN,NaN,0.4738,1.1147,1.0,1.0,4.0,1.0,4.0,0.0,0.0,0.0,1.0,0.0,0.0,NaN,0.0,0.0,1.0,1.0,1.0,NaN,NaN,1.0,1.0,1.0,NaN,NaN,1.0,3.0,11.0,0.0,0.0,6.0,5.0,0.0,3.0,109811.0,3060.2229,NaN,3964.1245,3855.4668,6077.0,128.0,3.5671,49.0,1.3655,62.0,258.0,7.1900,63490,40151.0,1118.9317,NaN,61263.0,-0.080853,0.28,NaN,883.0,214.0,NaN,NaN,30.5713,15638.0,18555.0,341.0,15683.0,18902.0,337.0,-45.0,-347.0,4.0,1.0,2.0,1.0,1.0,1.0,1.0,26596.0,29614.0,526.0,24546.0,30439.0,524.0,2050.0,-825.0,2.0,5.0,13.0,4.0,4.0,9.0,5.0,35843.0,41210.0,724.0,35989.0,44008.0,724.0,-146.0,-2798.0,0.0,7.0,19.0,9.0,9.0,26.0,7.0,44696.0,52664.0,839.0,45124.0,53573.0,860.0,-428.0,-909.0,-21.0,12.0,36.0,11.0,11.0,30.0,12.0
76031,ESPORTSTMNT02_2778045,complete,NaN,CBLOL,2022,Split 2,0,2022-06-11 15:46:13,1,12.1,200,Red,team,NaN,NaN,RED Canids,oe:team:67169969ecc058b307edfa9c6b7a3c0,0.0,NaN,Gwen,Wukong,Viego,Sylas,Lillia,Taliyah,Tristana,Alistar,Aatrox,Trundle,35.883333,1,23,17,65,23,17,2.0,2.0,1.0,0.0,0.0,NaN,NaN,NaN,0.6410,1.1147,0.0,4.0,1.0,4.0,1.0,0.0,0.0,1.0,0.0,0.0,3.0,NaN,0.0,0.0,0.0,1.0,1.0,NaN,NaN,0.0,1.0,1.0,NaN,NaN,0.0,11.0,3.0,1.0,1.0,5.0,6.0,3.0,0.0,102969.0,2869.5495,NaN,3984.0502,4334.0455,24205.0,116.0,3.2327,53.0,1.4770,57.0,284.0,7.9145,72190,48851.0,1361.3841,NaN,66425.0,0.080853,-0.28,NaN,928.0,272.0,NaN,NaN,33.4417,15683.0,18902.0,337.0,15638.0,18555.0,341.0,45.0,347.0,-4.0,1.0,1.0,1.0,1.0,2.0,1.0,24546.0,30439.0,524.0,26596.0,29614.0,526.0,-2050.0,825.0,-2.0,4.0,9.0,5.0,5.0,13.0,4.0,35989.0,44008.0,724.0,35843.0,41210.0,724.0,146.0,2798.0,0.0,9.0,26.0,7.0,7.0,19.0,9.0,45124.0,53573.0,860.0,44696.0,52664.0,839.0,428.0,909.0,21.0,11.0,30.0,12.0,12.0,36.0,11.0
76066,ESPORTSTMNT02_2778108,complete,NaN,CBLOL,2022,Split 2,0,2022-06-11 17:35:55,1,12.1,100,Blue,team,NaN,NaN,INTZ,oe:team:bcb21457414eb9be1d27489ff7e8e59,1.0,NaN,Yone,Renata Glasc,Ahri,Galio,Lillia,Taliyah,Viego,Xayah,Gnar,Rakan,34.450000,1,18,6,47,18,6,4.0,

##  Agregação por Partida Única e Resumo de Letalidade

Embora tenhamos isolado as linhas de equipa, cada confronto competitivo ainda ocupa **duas linhas** na tabela `df_filtrado` (uma linha para o sumário do time Blue e outra para o sumário do time Red).

Para analisarmos a letalidade real de um jogo completo, precisamos de agrupar essas duas linhas num registo único por `gameid`. Construímos um dicionário de agregação personalizado no `.groupby()` aplicando regras específicas para cada tipo de dado:
* `kills: 'sum'` ➡️ Soma os abates dos dois lados, gerando o total de mortes da partida.
* `gamelength: 'first'` ➡️ Como o tempo de jogo é idêntica para os dois lados, extraímos o primeiro valor de forma íntegra.
* `teamname: lambda x: ' vs '.join(x)` ➡️  Criamos uma função lambda que une o nome das duas equipas com um conector `" vs "`.

In [64]:
km = df_filtrado.groupby('gameid').agg({
    'kills': 'sum',
    'gamelength': 'first',
    'league' : 'first',
    'ckpm' : 'first',
    'teamname': lambda x: ' vs '.join(x),
}).copy()
km.reset_index(inplace = True)

In [65]:
km

,gameid,kills,gamelength,league,ckpm,teamname
0,ESPORTSTMNT02_2778045,40,35.883333,CBLOL,1.1147,paiN Gaming vs RED Canids
1,ESPORTSTMNT02_2778108,24,34.450000,CBLOL,0.6967,INTZ vs Rensga eSports
2,ESPORTSTMNT02_2778137,24,32.550000,CBLOL,0.7373,Liberty vs Miners
3,ESPORTSTMNT02_2778177,24,25.000000,CBLOL,0.9600,FURIA vs KaBuM! Esports
4,ESPORTSTMNT02_2778214,30,37.000000,CBLOL,0.8108,Flamengo MDL vs LOUD
...,...,...,...,...,...,...
474,ESPORTSTMNT06_2542204,25,29.050000,LCK,0.8606,Nongshim RedForce vs T1
475,ESPORTSTMNT06_2542219,23,34.050000,LCK,0.6755,Nongshim RedForce vs T1
476,ESPORTSTMNT06_2542222,35,34.833333,LCK,1.0048,T1 vs Nongshim RedForce
477,ESPORTSTMNT06_2542223,21,35.600000,LCK,0.5899,DN SOOPers vs Kiwoom DRX


##  Análise de Ritmo: Índice de Abates Combinados por Minuto (CKPM)

Com a nossa base consolidada por partida única na tabela `km`, iniciamos a fase de análise visual avaliando o **CKPM** (*Combined Kills Per Minute*). Essa métrica calcula a taxa de letalidade ativa do jogo através da fórmula:

$$CKPM = \frac{\text{Abates Totais da Partida}}{\text{Duração da Partida em Minutos}}$$

**Nota Estatística sobre a Métrica:** Como o CKPM já é uma taxa calculada por partida, cada linha da nossa tabela já guarda o ritmo médio daquele confronto específico. No código abaixo, utilizamos o `px.strip` (gráfico de dispersão unidimensional) para plotar **cada uma das partidas como um ponto individual**. O gráfico não calcula uma média matemática no código; ele expõe a distribuição bruta de todos os jogos, permitindo avaliar visualmente onde os dados se concentram e a frequência de casos extremos (*outliers*).

In [66]:
fig = px.strip(
    data_frame=km.reset_index(),
    x='league',
    y='ckpm',
    color='league',
    hover_data=['gameid', 'kills', 'gamelength'],
    title='Distribuição de Cada Jogo por Liga'
)

fig.update_layout(template='plotly_white', xaxis_title='Liga', yaxis_title='CKPM')
fig.show()

###  Análise do Gráfico: A Assinatura de Ritmo de Cada Região

A distribuição vertical e a densidade das "nuvens" de pontos revelam assinaturas culturais muito claras para cada uma das regiões estudadas:

* **CBLOL (Nuvem Azul):** Exibe uma forte concentração de pontos deslocada para o topo do gráfico, orbitando de forma densa entre as faixas de **0.8 e 1.2 CKPM**. Isso comprova que o ecossistema brasileiro possui uma cultura de jogo voltada para lutas constantes e agressões contínuas no mapa.
* **LCK (Nuvem Laranja):** Mostra um comportamento inversamente oposto, com seus pontos fortemente concentrados na base inferior do gráfico (faixa de **0.4 a 0.7 CKPM**). Esse desenho quantifica o estilo coreano: um jogo cirúrgico, focado em controle tático, punição de erros e negação sistemática de riscos. Os poucos pontos isolados no topo provam que, quando a estratégia inicial falha, a LCK também experiência partidas caóticas.
* **LEC (Nuvem Verde):** Apresenta uma distribuição vertical muito mais uniforme e espalhada. Isso indica que a Europa não possui uma identidade de ritmo tão homogênea; a letalidade flutua livremente dependendo da preferência individual dos times.

##  Gerenciamento de Tempo: Duração Média das Partidas

Compreendido o ritmo de abates, a próxima pergunta essencial é: *Ligas mais agressivas destroem o Nexus mais rápido?* Para responder a isso, calculamos o tempo médio de duração das partidas (`gamelength`) por liga. No gráfico gerado a seguir, aplicamos um corte de visualização no eixo Y entre **30 e 34 minutos**. Esse "zoom" visual ajuda a evidenciar as sutis diferenças de minutos e segundos que separam o encerramento das partidas em cada ecossistema.

In [67]:
tempo_partida = df_filtrado.groupby('league').agg({
    'gamelength': 'mean'
}).copy()
tempo_partida.reset_index(inplace = True)

In [68]:
tempo_partida

,league,gamelength
0,CBLOL,32.778650
1,LCK,33.118652
2,LEC,33.594851


In [69]:
fig = px.bar(
    data_frame=tempo_partida,
    x='league',
    y='gamelength',
    color='league',
    title='Comparação Direta de Tempo de Jogo por Liga'
)


fig.update_layout(
    xaxis_title='Liga',
    yaxis_title='Tempo Total (Escala da Tabela)',
    template='plotly_white'
)
fig.update_yaxes(range=[30, 34])

fig.show()

### Análise do Gráfico: O Mito da Duração de Jogo Desmistificado

A visualização traz uma revelação valiosa: **as partidas profissionais duram quase exatamente o mesmo tempo no mundo inteiro**, orbitando muito próximo da barreira dos **33 minutos**.

* O **CBLOL** é sutilmente a liga mais rápida na média (**32.7 min**), enquanto a **LEC** é a mais demorada (**33.5 min**).
* **O Insight Estratégico:** Cruzando esse resultado com o gráfico anterior de CKPM, chegamos a uma grande conclusão tática. A real divergência entre o LoL coreano (LCK) e o brasileiro (CBLOL) não está no relógio final do jogo. A diferença é sobre **como esses 33 minutos são preenchidos**: enquanto o Brasil gasta o tempo acumulando abates e lutas, a Coreia do Sul gasta os mesmos 33 minutos em rotações meticulosas, controle de visão e cerco a objetivos neutros.

##  A Guerra de Visão: Wards por Minuto (WPM) vs Placar de Visão (VSPM)

Para aprofundar o entendimento sobre o controle estratégico de cada região, isolamos duas métricas fundamentais de mapa: `wpm` (quantidade de sentinelas colocadas por minuto) e `vspm` (placar geral de visão obtido por minuto pelas equipes).

 Geramos um gráfico de barras agrupadas para colocar essas duas métricas lado a lado por liga, permitindo avaliar não apenas o volume de recursos de visão investidos, mas a eficiência real desse investimento.

In [70]:
vision_per_min = df_filtrado.groupby('league').agg({
    'wpm': 'mean',
    'vspm' : 'mean',

}).copy()
vision_per_min.reset_index(inplace = True)

In [71]:
vision_per_min

,league,wpm,vspm
0,CBLOL,3.506357,7.595024
1,LCK,3.540343,7.928190
2,LEC,3.219990,7.457540


In [72]:
fig = px.bar(
    data_frame=vision_per_min,
    x='league',
    y=['wpm', 'vspm'],
    barmode='group',
    title='Guerra de Visão: Wards por Minuto (WPM) vs Placar de Visão por Minuto (VSPM)'
)


fig.update_layout(
    xaxis_title='Liga',
    yaxis_title='Média por Minuto',
    legend_title='Métricas de Visão',
    template='plotly_white'
)

fig.show()

### 💡 Análise do Gráfico: Eficiência Tática na Iluminação do Mapa

Os dados revelam um padrão fascinante sobre a eficiência do controle de mapa:

* **O Volume de Investimento:** Note que o **CBLOL (3.50 WPM)** e a **LCK (3.54 WPM)** colocam praticamente a mesma quantidade de sentinelas no chão a cada minuto de jogo.
* **A Diferença na Eficiência:** Embora depositem o mesmo número de wards no mapa, a LCK se descola na liderança com o **VSPM de 7.92** em relação ao CBLOL (7.59). Como o Placar de Visão pontua a negação de wards inimigas e a eficácia em detectar oponentes, esse gráfico prova que os times coreanos extraem muito mais inteligência tática do mesmo número de wards colocadas, posicionando-as nos locais mais cirúrgicos possíveis.

##  Dinâmica de Early Game: Vantagem aos 15 Minutos dos Vencedores

Por fim, investigamos a capacidade de transição e resiliência de cada liga. Filtramos a base de dados para isolar apenas as equipes que **venceram as partidas (`result == 1`)** e avaliamos qual era o saldo de diferença de ouro que elas possuíam na marca exata dos 15 minutos (`golddiffat15`) — ponto que marca o fim da fase de rotas.

Utilizamos um **Boxplot com pontos sobrepostos** para identificar a mediana dessas vantagens e mapear a ocorrência de grandes reviravoltas. A linha horizontal cinza fixada no ponto **ZERO** serve como referência de empate absoluto aos 15 minutos; logo, **qualquer ponto plotado no quadrante negativo representa um "Comeback" (Vira-Vira)**.

In [73]:
df_filtrado[['golddiffat15' , 'result' , 'teamname' , 'gameid' , 'league']].head(6)

,golddiffat15,result,teamname,gameid,league
76030,2050.0,0,paiN Gaming,ESPORTSTMNT02_2778045,CBLOL
76031,-2050.0,1,RED Canids,ESPORTSTMNT02_2778045,CBLOL
76066,1573.0,1,INTZ,ESPORTSTMNT02_2778108,CBLOL
76067,-1573.0,0,Rensga eSports,ESPORTSTMNT02_2778108,CBLOL
76090,-1405.0,1,Liberty,ESPORTSTMNT02_2778137,CBLOL
76091,1405.0,0,Miners,ESPORTSTMNT02_2778137,CBLOL


In [74]:
vant_15_min = df_filtrado[df_filtrado['result'] == 1] [['golddiffat15' , 'teamname' , 'league' ,  'result' ]].copy().reset_index()

In [75]:
fig = px.box(
    data_frame=vant_15_min,
    x='league',
    y='golddiffat15',
    color='league',
    points='all',
    hover_data=['teamname'],
    title='Vantagem de Ouro aos 15 Minutos dos Times Vencedores'
)


fig.add_hline(y=0, line_dash="dash", line_color="gray", annotation_text="Jogo Empatado aos 15 min")

fig.update_layout(
    xaxis_title='Liga',
    yaxis_title='Diferença de Ouro aos 15 min (golddiffat15)',
    template='plotly_white'
)

fig.show()

### 💡 Análise do Gráfico: Imprevisibilidade Caótica vs Rigidez Macro

O Boxplot expõe perfeitamente o nível de estabilidade macroeconômica de cada ecossistema:

* **CBLOL -  A caixa azul abraça a linha do zero e estende raízes profundas no quadrante negativo. O CBLOL possui uma nuvem densa de partidas abaixo da linha de empate, incluindo um caso extremo (*outlier*) no índice `103594`, onde a equipe da **Miners** conseguiu vencer a partida mesmo enfrentando uma desvantagem colossal de **-5.245 de ouro aos 15 minutos** contra a RED Canids. Isso prova que o início de jogo no Brasil não dita o resultado final; a liga é altamente imprevisível e propensa a reviravoltas.
* **LCK - Em total contraste, a nuvem laranja da Coreia do Sul é extremamente magra abaixo da linha de zero, com quase nenhum ponto ultrapassando a barreira dos -2.000 de ouro. Isso atesta o controle macro da LCK: quando um time coreano fecha a fase de rotas em vantagem financeira, ele joga de forma tão segura e protocolar que o adversário raramente encontra brechas para uma virada. Quem fica rico cedo na LCK, geralmente vence de forma limpa.
* **LEC - A Europa se posiciona como uma zona de transição equilibrada. Suas caixas estão consolidadas acima do zero, indicando previsibilidade, mas a região ainda apresenta uma quantidade moderada de reviravoltas, sem atingir os extremos caóticos vistos no Brasil.